serde/src/java/org/apache/hadoop/hive/serde2/lazybinary/objectinspector/LazyBinaryObjectInspectorFactory.java

```diff
@@ -53,10 +54,10 @@ public static LazyBinaryStructObjectInspector getLazyBinaryStructObjectInspector
   public static LazyBinaryStructObjectInspector getLazyBinaryStructObjectInspector(
       List<String> structFieldNames,
       List<ObjectInspector> structFieldObjectInspectors, List<String> structFieldComments) {
-    ArrayList<Object> signature = new ArrayList<Object>(3);
-    signature.add(structFieldNames);
-    signature.add(structFieldObjectInspectors);
-    if(structFieldComments != null) {
+    List<Object> signature = structFieldComments == null ? Arrays.asList(structFieldNames, structFieldObjectInspectors)
+        : Arrays.asList(structFieldNames, structFieldObjectInspectors, structFieldComments);
+
+    if (structFieldComments != null) {
```

Comment:
Isn't this a duplication? I mean if structFieldComments is not null, it will be already in signature due to the command above.

Reply:
Yup! That's the code I was trying to replace, but didn't actually remove it.  Thanks!